src/__tests__/pages/Research.test.js

```diff
@@ -0,0 +1,198 @@
+    image: "uk-triple-lock-plus.jpg",
+  },
+];
+
+describe("Research Page", () => {
+  test("renders without crashing", () => {
+    render(
+      <BrowserRouter>
+        <Research />
+      </BrowserRouter>,
+    );
+
+    const heading = screen.getByRole("heading", {
+      name: /Research and analysis/i,
+    });
+    expect(heading).toBeInTheDocument();
+
+    const subheadingText =
+      "Read PolicyEngine's research on recent and proposed policy reforms, as well as technical and general updates from the organisation.";
+    const subheading = screen.getByText(subheadingText);
+    expect(subheading).toBeInTheDocument();
+  });
+});
+
+describe("BlogPostResults Component", () => {
+  test("renders the correct number of blog posts", () => {
+    render(
+      <BrowserRouter>
+        <BlogPostResults posts={blogPosts} />
+      </BrowserRouter>,
+    );
+
+    blogPosts.forEach((post) => {
+      const titleElement = screen.getByText(post.title);
+      expect(titleElement).toBeInTheDocument();
+
+      const descriptionElement = screen.getByText(post.description);
+      expect(descriptionElement).toBeInTheDocument();
+    });
+  });
+
+  test("renders blog post links", () => {
+    render(
+      <BrowserRouter>
+        <BlogPostResults posts={blogPosts} />
+      </BrowserRouter>,
+    );
+
+    blogPosts.forEach((post) => {
+      const titleElement = screen.getByText(post.title);
+      expect(titleElement).toBeInTheDocument();
+
+      const descriptionElement = screen.getByText(post.description);
+      expect(descriptionElement).toBeInTheDocument();
+
+      // Extract filename without extension
+      const filenameWithoutExtension = post.filename.replace(/\.[^.]+$/, "");
+
+      // Find the "Read" link associated with the current blog post
+      const readLinks = screen.getAllByText("Read");
+      const link = readLinks.find((readLink) => {
+        const parent = readLink.closest(".stretched-link");
+        return (
+          parent &&
+          parent.getAttribute("href") ===
+            `/null/research/${filenameWithoutExtension}`
+        );
+      });
+
+      expect(link).toBeInTheDocument();
+    });
+  });
+});
+describe("MediumBlogPreview Component", () => {
+  test("renders blog post details", () => {
+    blogPosts.forEach((mockPost, index) => {
+      render(
+        <BrowserRouter key={index}>
+          <MediumBlogPreview key={mockPost.title} blog={mockPost} />
+        </BrowserRouter>,
+      );
+
+      screen.debug();
+
+      const titleElement = screen.getByText(mockPost.title);
+      expect(titleElement).toBeInTheDocument();
+
+      const descriptionElement = screen.getByText(mockPost.description);
+      expect(descriptionElement).toBeInTheDocument();
+
+      // Check if at least one "Read" link exists
+      const readLinks = screen.getAllByText("Read");
+      const readLink = readLinks.find(
+        (link) =>
+          link.closest(".stretched-link")?.getAttribute("href") ===
```

Comment:
This is good, but to avoid breaking if Ant Design ever changes their CSS styles, could you instead use `screen.getAllByRole("link", {name: /READ arrow_forward/i})` to get the links, then use their `value` property to check the href? If you want to view all the properties available to you, if you right-click on an element on our page and click something like "View accessibility properties," that may help to give ideas for details to match against.